multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/AdditionalModuleParametersReporter.java

```diff
@@ -0,0 +1,54 @@
+package org.cloudfoundry.multiapps.controller.process.util;
+
+import java.text.MessageFormat;
+import java.util.List;
+
+import org.cloudfoundry.multiapps.controller.core.model.SupportedParameters;
+import org.cloudfoundry.multiapps.controller.process.steps.ProcessContext;
+import org.cloudfoundry.multiapps.controller.process.variables.Variables;
+import org.cloudfoundry.multiapps.mta.model.Module;
+import org.cloudfoundry.multiapps.mta.util.PropertiesUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AdditionalModuleParametersReporter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AdditionalModuleParametersReporter.class);
+
+    private final ProcessContext context;
+
+    public AdditionalModuleParametersReporter(ProcessContext context) {
+        this.context = context;
+    }
+
+    public void reportUsageOfAdditionalParameters(Module module) {
+        String mtaId = context.getRequiredVariable(Variables.DEPLOYMENT_DESCRIPTOR)
```

Comment:
Is it possible to not have Variables.DEPLOYMENT_DESCRIPTOR here?
E.g. rollback, undeploy or transport scenarios

Reply:
No, it is set in ProcessArchiveStep in normal processes including transports, and in PrepareBackupMtaForDeploymentStep during rollback. Not used during undeploy